com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/usecases/TestVariableUsedSeveralTimesInOneStatement.java

```diff
@@ -0,0 +1,115 @@
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This test checks that document with multi-definitions parsed correctly and all the usages of
+ * variables placed in one line found.
+ */
+public class TestVariableUsedSeveralTimesInOneStatement {
+  private static final String TEXT =
+      "       Identification Division. \n"
+          + "       Program-id.    ProgramId.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01   PARENT1. \n"
+          + "        02   CHILD      PIC 9(4) Binary. \n"
+          + "       01   PARENT2.\n"
+          + "        02   CHILD      PIC 9(4) Binary. \n"
+          + "       Procedure Division.\n"
+          + "       000-Main-Logic.\n"
+          + "           MOVE CHILD IN PARENT1 TO CHILD IN PARENT2.\n"
+          + "       End program ProgramId.";
+
+  private static final String CHILD = "CHILD";
+  private static final String PARENT1 = "PARENT1";
+  private static final String PARENT2 = "PARENT2";
+  private static final String MAIN_LOGIC = "000-MAIN-LOGIC";
+
+  @Test
+  public void test() {
+    AnalysisResult result = analyze(TEXT);
+    assertDiagnostics(result.getDiagnostics());
+
+    assertCopybookUsages(result.getCopybookUsages());
+    assertCopybookDefinitions(result.getCopybookDefinitions());
+
+    assertVariableUsages(result.getVariableUsages());
+    assertVariableDefinitions(result.getVariableDefinitions());
+
+    assertParagraphDefinitions(result.getParagraphDefinitions());
+    assertParagraphUsages(result.getParagraphUsages());
+  }
+
+  private void assertDiagnostics(Map<String, List<Diagnostic>> diagnostics) {
+    long numberOfDiagnostics = diagnostics.values().stream().mapToLong(List::size).sum();
+    assertEquals("Diagnostics: " + diagnostics.toString(), 0, numberOfDiagnostics);
+  }
+
+  private void assertParagraphUsages(Map<String, List<Location>> usages) {
+    assertEquals("Paragraph usages: " + usages.toString(), 0, usages.size());
+  }
+
+  private void assertParagraphDefinitions(Map<String, List<Location>> definitions) {
+    assertEquals("Paragraph definitions: " + definitions.toString(), 1, definitions.size());
+    assertNumberOfLocations(definitions, MAIN_LOGIC, 1);
+    assertLocation(definitions, MAIN_LOGIC, DOCUMENT_URI, 9, 7);
+  }
+
+  private void assertCopybookDefinitions(Map<String, List<Location>> copybookDefinitions) {
+    assertEquals(
+        "Copybook definitions: " + copybookDefinitions.toString(), 0, copybookDefinitions.size());
+  }
+
+  private void assertCopybookUsages(Map<String, List<Location>> copybookUsages) {
+    assertEquals("Copybook usages: " + copybookUsages.toString(), 0, copybookUsages.size());
+  }
+
+  private void assertVariableDefinitions(Map<String, List<Location>> definitions) {
+    assertEquals("Variable definitions: " + definitions.toString(), 3, definitions.size());
+
+    assertNumberOfLocations(definitions, PARENT1, 1);
```

Comment:
In many tests, there are similar/same testing steps with different value. 
It would be perfect to extract it to a single method.

Reply:
I agree, but I didn't find a way how to do it handily

Reply:
Mostly similar steps can be extracted into method via parameters, actually you have used such approaches already.

Reply:
I'll suggest planning the activity regarding tests refactoring, maybe on the innovation week, because it is not the only problem here.

Reply:
I am preparing some topic regarding this theme because I am focusing on testing for copybook replace in extended document.
I skipped the static analysis of the code, but if u think that I could contribute on that as well let me know.

Reply:
#427